cli/internal/cache/cache_http.go

```diff
@@ -276,9 +277,10 @@ func restoreTar(root turbopath.AbsoluteSystemPath, reader io.Reader) ([]string,
 			return nil, err
 		}
 		// hdr.Name is always a posix-style path
-		// TODO: files should eventually be repo-relative system paths
-		files = append(files, hdr.Name)
-		filename := root.UntypedJoin(hdr.Name)
+		// FIXME: THIS IS A BUG.
+		restoredName := turbopath.AnchoredUnixPath(hdr.Name)
```

Comment:
There are cases where this is a bug. This will be addressed in the cache-to-tar portion.